src/SFA.DAS.EAS.Web.IntegrationTests/TestUtils/DataHelper/EmployerAccountsDbBuilder.cs

```diff
@@ -33,31 +33,38 @@ class DbBuilder
 
         public DbBuilderDependentRepositories DependentRepositories { get; }
 
-        public DbBuilder BeginTransaction()
+        public bool HasTransaction()
         {
-            _dbContext.Database.BeginTransaction();
+            return (_dbContext.Database.CurrentTransaction != null);
+        }
+
+        public EmployerAccountsDbBuilder EnsureTransaction()
+        {
+            if(_dbContext.Database.CurrentTransaction == null)
+                _dbContext.Database.BeginTransaction();
+
             return this;
         }
 
-        public DbBuilder CommitTransaction()
+        public EmployerAccountsDbBuilder CommitTransaction()
         {
             if (_dbContext.Database.CurrentTransaction == null) return this;
```

Comment:
If someone calls commit and the request can't be honoured then something has gone wrong. Should we be silently ignoring this or might it better to throw an exception do you think?

Reply:
Method is obsolete - I have removed.